src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/profiler/ProfilerEditingTarget.java

```diff
@@ -378,6 +375,26 @@ public void revertChanges(Command onCompleted)
    {
       onCompleted.execute();
    }
+   
+   private String getAndSetInitialName()
```

Comment:
I thought about this for a second but noticed that the `nameProvider` is reused by most documents and the convention seems to be `Untitled#` with an icon for the respective document type. However, I do agree `Profiler#` is more clear; will make the change on separate PR.